### PR TITLE
prune: don't remove paths ending in .d

### LINKED ISSUE #31760: mariadb: missing /usr/local/etc/my.cnf.d, removed by brew prune
When installing MariaDB and trying to start things via `mysql.server start` the following error is returned:

```
mysql.server start
/usr/local/Cellar/mariadb/10.0.13/bin/my_print_defaults: Can't read dir of '/usr/local/etc/my.cnf.d' (Errcode: 2 "No such file or directory")
Fatal error in defaults handling. Program aborted
Starting MySQL
. ERROR!
```

Debug info: https://gist.github.com/ActionScripted/cc9f6b83f0b55239833b

The fix is to manually create the directory:

```
mkdir /usr/local/etc/my.cnf.d
```

And things work great! However if you run `brew prune` it removes this directory and you have to manually create it again. I'm not sure if the issue is `brew prune` dropping an empty directory or `mkpath` in the formula removing things or what.

I've completely removed mariadb, cleaned things up and reinstalled to see if maybe something got weird on my system but it's always the same issue. If there's anything I can do to help or any other information I can contribute that would be useful I'd be glad to help.

### PR DESCRIPTION
**Pull request by tdsmith**
These may have semantic content. Closes #31760.

### CODE REVIEW

**Comment by tdsmith** [comment]
Alternatively we could just `touch .homebrew_dont_prune_me` in directories where this matters.

**Comment by mikemcquaid** [comment]
Might want to be more specific about this change and create it in postinstall (if that works) or whitelist a more specific path. Also wants a comment, probably.

File: Library/Homebrew/cmd/prune.rb [hunk 1]
```diff
@@ -29,6 +29,7 @@ def prune
     end
 
     dirs.reverse_each do |d|
+      next if d.basename.to_s.end_with? ".d"
```
Comment (jacknagel) [inline Library/Homebrew/cmd/prune.rb]: `d.extname == ".d"`?

Comment (tdsmith) [inline Library/Homebrew/cmd/prune.rb]: Oops, thanks.

**Comment by tdsmith** [comment]
I think I like the `touch` option better unless you prefer this.

**Pull request closed without merging**